README.md

```diff
@@ -345,22 +345,64 @@ The wrapper supports rust installed via [`rustup`](https://rustup.rs/)
 
 ```dockerfile
 # syntax=docker/dockerfile:1
-# official rust image
 FROM --platform=$BUILDPLATFORM rust:alpine
 RUN apk add clang lld
 # ...
-RUN --mount=type=cache,target=/usr/local/cargo/git/db \
-    --mount=type=cache,target=/usr/local/cargo/registry/cache \
-    --mount=type=cache,target=/usr/local/cargo/registry/index \
-    cargo fetch
 ARG TARGETPLATFORM
 RUN xx-cargo build --release --target-dir ./build && \
-    xx-verify ./build/$(xx-cargo --print-target)/release/hello_cargo
+    xx-verify ./build/$(xx-cargo --print-target-triple)/release/hello_cargo
+```
+
+Cargo binary can also be called directly with `--target` flag if you don't want
+to use the wrapper. `--print-target-triple` is a built-in flag that can be used
+to set the correct target:
+
+```dockerfile
+# syntax=docker/dockerfile:1
+FROM --platform=$BUILDPLATFORM rust:alpine
+RUN apk add clang lld
+# ...
+ARG TARGETPLATFORM
+RUN cargo build --target=$(xx-cargo --print-target-triple) --release --target-dir ./build && \
+    xx-verify ./build/$(xx-cargo --print-target-triple)/release/hello_cargo
 ```
 
+> **Note**
+>
+> `xx-cargo --print-target-triple` does not always have the same value as
+> `xx-clang --print-target-triple`. This is because prebuilt Rust and C
+> libraries sometimes use a different value.
+
+The first invocation of `xx-cargo` will install the standard library for Rust
+matching the target if not already installed.
+
+To fetch dependencies from crates.io you can use `cargo fetch` before building:
```

Comment:
> before building

Add that "By calling `cargo fetch` before `ARG TARGETPLATFORM` your packages are fetched only once for the whole build while the building happens separately for each target architecture."